nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/ClusterProtocolHeartbeatMonitor.java

```diff
@@ -91,12 +91,8 @@ public ClusterProtocolHeartbeatMonitor(final ClusterCoordinator clusterCoordinat
         this.clusterNodesPath = zkClientConfig.resolvePath("cluster/nodes");
 
         String hostname = properties.getProperty(NiFiProperties.CLUSTER_NODE_ADDRESS);
-        if (hostname == null) {
-            try {
-                hostname = InetAddress.getLocalHost().getHostName();
-            } catch (UnknownHostException e) {
-                throw new RuntimeException("Unable to determine local hostname and the '" + NiFiProperties.CLUSTER_NODE_ADDRESS + "' property is not set");
-            }
+        if (hostname == null || hostname.trim().isEmpty()) {
+            hostname = "localhost";
```

Comment:
Is there any specific reason to change from `InetAddress.getLocalHost().getHostName()` to `"localhost"`?

I thought if server's hostname is properly configured at OS level, administrator doesn't have to set hostname in nifi.properties, which makes configuration management easier since they can use the same properties file among nodes.